first pass fileinput (#1565)
grommet/hpe-design-system: aries-site/src/pages/components/fileinput.mdx

import {
  FileInputSimpleExample,
  FileInputSingleExample,
  FileInputMultipeExample,
} from '../../examples/components/fileinput';

<Example
  docs="https://v2.grommet.io/fileinput?theme=hpe"
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/fileinput/FileInputSimpleExample.js"
  figma="https://www.figma.com/file/8HwAiiSDloOTjUxqvCTypq/HPE-FileInput-Component"
>
  <FileInputSimpleExample />
</Example>

## Guidance

FileInput is a specialized input that provides a clean interface for selecting files.
FileInput can be used when a user needs to upload one or more files.

### Usage

1. Upload a file or files by click on the fileInput that prompts a file selection.
2. Drag and Drop selected files directly into the fileInput.
3. Label the fileInput so the user knows what file/files needs to be uploaded

### Accessibility

The FileInput can be used with both the `Space` and `Enter` button to achieve the action.
FileInput should have clear label that describes the type of files that are needed.

## Variants

FileInputs can be used as a single file upload or multiple files.

### FileInput with a single file

> <Example code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/fileinput/FileInputSingleExample.js">

Reviewer: We should add the `docs` prop and link to grommet site docs for FileInput.

  <FileInputSingleExample />
</Example>

### FileInput with multiple files

> <Example code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/fileinput/FileInputMultipeExample.js">

Reviewer: We should add the `docs` prop and link to grommet site docs for FileInput.

import { Example } from '../../layouts';
import { ExampleImagePreview } from '../../components';
import {
  FileInputSimpleExample,
  FileInputMultipleExample,
} from '../../examples/components/fileinput';

<Example
  docs="https://v2.grommet.io/fileinput?theme=hpe"
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/fileinput/FileInputSimpleExample.js"
  figma="https://www.figma.com/file/8HwAiiSDloOTjUxqvCTypq/HPE-FileInput-Component"
>
  <FileInputSimpleExample />
</Example>

## Guidance

> FileInput is a specialized input that provides the user with a means of selecting a file or files for upload.

Reviewer 2: ```suggestion
FileInput is a specialized input providing users a means to select a file or files for upload.
```

### Usage

> - User must be able to upload a file by clicking the "Select File" button, drag and dropping over the drop region, or tabbing to the FileInput and pressing Space/Enter.

Reviewer 2: ```suggestion
- User must be able to upload a file by clicking the "Select File" button, dragging and dropping a file over the drop region, or tabbing to the FileInput and pressing the Spacebar or Enter key.
```

- When FileInput is in a Form, the file should not be uploaded/saved to a server until the user has submitted the form.

> - Once a file has been added to the input, a Trash button will appear next to the file name to allow removal of that file.

Reviewer 2: ```suggestion
- Once a file has been added to the input, a button featuring a Trash icon will appear next to the file name allowing removal of that file.
```

- The "Select File" button should always be visible, allowing the user to add more files/replace a file after they have already selected one.

For additional configurations, such as defining accepted file type, all instrinsic HTML properties are supported. For a comprehensive list of these options, you can read more about [HTML input type="file"](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file).

### Accessibility

> When using FileInput within a Form, the FormField wrapping the FileInput should have `htmlFor` and `name` properties that match the `id` and `name` properties placed onto the FileInput. See the [FileInput within a Form example](/templates/forms#required-and-optional-fields) for reference on how to implement.

Reviewer 2: ```suggestion
When using FileInput within a Form, the FormField wrapping the FileInput should have `htmlFor` and `name` properties that match the `id` and `name` properties placed onto the FileInput. Reference the [FileInput within a Form example](/templates/forms#required-and-optional-fields) to see how to implement.
```

## Variants

> FileInputs can be used as a single file upload or multiple files.

Reviewer 2: ```suggestion
FileInputs may be configured for uploading single or multiple files.
```
